activerecord/lib/active_record/connection_adapters/connection_specification.rb

```diff
@@ -17,6 +17,37 @@ def to_hash
         @config.merge(name: @name)
       end
 
+      class ConnectionConfigurations < SimpleDelegator #:nodoc:
+        attr_accessor :root_level
+
+        def initialize(config)
+          super
+          @root_level = nil
+        end
+
+        def at(top_level)
+          normalized(top_level)
+        end
+
+        def normalized(key = @root_level)
+          config = key ? self[key] : self
+          config = config ? config.dup : {}
```

Comment:
Find these two lines hard to parse because of the two conditionals. The empty hash fallback seems too defensive at first sight. What case is it trying to cover? That there's a key in `self[key]` but with no config at all?

Could we simplify to this?

```ruby
config = fetch(key, self).dup
```